### PR TITLE
Refactor docker-py compatibility tests

### PR DESCRIPTION
**Pull request by jwhonce**
* Add which python client is being used to run tests, see "python
  client" below.
* Remove redundant code from test classes
* Update/Add comments to modules and classes

======================================================= test session starts ========================================================
platform linux -- Python 3.10.0, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
python client -- DockerClient
rootdir: /home/jhonce/Projects/go/src/github.com/containers/podman
plugins: requests-mock-1.8.0
collected 33 items

test/python/docker/compat/test_containers.py ...s..............  [ 54%]
test/python/docker/compat/test_images.py ............  [ 90%]
test/python/docker/compat/test_system.py ...  [100%]

Note: Follow-up PRs will verify the test results and expand the tests.

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by jwhonce** [comment]
@cevich This will add a header to each test report with the marker `python client -- DockerClient`, I will update the other python tests with appropriate markers. I also took the liberty to clean up the code.

**Comment by edsantiago** [comment]
Logs: [apiv2](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6234396700180480/html/apiv2-podman-fedora-35-root-host.log.html) (colorized/formatted), [docker](https://api.cirrus-ci.com/v1/task/6691793537335296/logs/main.log) (plaintext). @jwhonce as long as you're poking in this area, is there any way you could get rid of the warnings in the apiv2 log? Those disrupt the output stream, causing `ok` to show up on empty lines, which makes my logformatter not see them (and therefore display a big nasty red "Expected NNNN" in the summary line). Not a huge deal, just an it-would-be-nice-to-fix.

**Comment by cevich** [comment]
Thank you so much @jwhonce I haven't looked yet, but I really appreciate the effort.  This kind of positive confirmation is really important long-term, especially with such a multifaceted matrix.  It's not glamorous work, but will surely lead to at least a few smaller bald-spots on developer heads :grin:

**Comment by cevich** [comment]
> causing ok to show up on empty lines

@edsantiago this may be really difficult to fix 100%.  Any output from the tests or test-subject happens essentially asynchronous with any test output.  Worse, the unittest system sits outside the standard (python) logging system - to support assertions against log output.  While it's possible to do some capturing of low-level output with `contextlib.redirect_stdout()` and `contextlib.redirect_stderr()`, the `unittest` output control interface is really wonky (last I checked several years ago).  Though perhaps Jhon knows better/easier way, it may be more practical to simply run in verbose-mode.

File: test/python/docker/compat/common.py [hunk 1]
```diff
@@ -1,23 +1,92 @@
...
+        time.sleep(2)
+
+        return_code = cls.service.poll()
+        if return_code is not None:
+            raise subprocess.CalledProcessError(return_code, "podman system service")
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.service.terminate()
+        stdout, stderr = cls.service.communicate(timeout=0.5)
```
Comment (cevich) [inline test/python/docker/compat/common.py:70]: Non-blocker:  This seems like a place where a hung sub-processes could leak.  Does it makes sense to catch the possible `TimeoutExpired` and `cls.service.kill()`?

**Comment by jwhonce** [comment]
@edsantiago For the localunit target, I disabled the warnings.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[tool.black]
+line-length = 100
+target-version = ['py36']
+include = '\.pyi?$'
+exclude = '''
+/(
+    \.git
+  | \.tox
+  | \.venv
```
[the remark this answers is not in the document]
Comment (jwhonce) [reply pyproject.toml:9]: .venv is the default for tox, I just copied and pasted. Renaming would be great. Thanks.

File: test/python/docker/compat/common.py (hunk 1 above)
Comment (jwhonce) [reply test/python/docker/compat/common.py:70]: @cevich kill() is an alias for terminate() at least in later python versions. Given we are cleaning up, I just went for the big hammer.

File: test/python/docker/__init__.py [hunk 3]
```diff
@@ -93,8 +91,8 @@ def __init__(self):
             }
             """
         )
-        with open(cni_cfg, "w") as w:
-            json.dump(buf, w)
+        with open(cni_cfg, "w") as file:
+            json.dump(buf, file)
```
[the remark this answers is not in the document]
Comment (jwhonce) [reply test/python/docker/__init__.py:95]: Sort answer: Yes.

We should write a card in to make the tests either OS-aware or network configuration neutral. The idea of building these configurations was to isolate the tests from developer laptop configurations.

An implementation could be use to fixtures to isolate and target the network configurations by network stack.

File: pyproject.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (cevich) [reply pyproject.toml:9]: ok

File: test/python/docker/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cevich) [reply test/python/docker/__init__.py:95]: Roger, I'll open a card, thanks.

[the remark this answers is not in the document]
Comment (cevich) [reply test/python/docker/__init__.py:95]: https://issues.redhat.com/browse/RUN-1456

**Comment by cevich** [comment]
LGTM

**Pull request merged**